spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/ConnectionFactoryUtils.java

```diff
@@ -167,18 +167,19 @@ public static void releaseResources(RabbitResourceHolder resourceHolder) {
 		RabbitUtils.closeConnection(resourceHolder.getConnection());
 	}
 
-	public static void bindResourceToTransaction(RabbitResourceHolder resourceHolder,
+	public static RabbitResourceHolder bindResourceToTransaction(RabbitResourceHolder resourceHolder,
 			ConnectionFactory connectionFactory, boolean synched) {
 		if (TransactionSynchronizationManager.hasResource(connectionFactory)
 				|| !TransactionSynchronizationManager.isActualTransactionActive() || !synched) {
-			return;
+			return (RabbitResourceHolder) TransactionSynchronizationManager.getResource(connectionFactory);
```

Comment:
I think it's only possible with weird configuration - such as...

    container.setTransactionAttribute(new DefaultTransactionAttribute(TransactionDefinition.PROPAGATION_NEVER));

...but possible - good catch.